### PR TITLE
fix idp schema, add virtual setup

### PR DESCRIPTION
**Pull request by butonic**
This just adds a config for the virtual views. The local integration tests now have two storage providers mounted at `/virtual` with the local driver. They can be accessed via the frontent-global service.

### CODE REVIEW

**Comment by butonic** [comment]
ok, I am now checking if the etage of `/`, `/virtual` and `/virtual/b` changes when the folder `/virtual/b/bar` is created.

Unfortunately, not even this basic test works, I assume becaus I am using the local driver which
1. did not do a propagation when creating a dir (fixed in this PR)
2. does not seem provide an etag for the root of the storage ... or I am not fully understanding how the stat and list across providers code works...

when doing a PROPFIND against the virtual folder I should see the etag thet corresponds to the later provider entry, like this:
```
diff --git a/internal/grpc/services/gateway/storageprovider.go b/internal/grpc/services/gateway/storageprovider.go
index dcfd7be3..5292ac30 100644
--- a/internal/grpc/services/gateway/storageprovider.go
+++ b/internal/grpc/services/gateway/storageprovider.go
@@ -1334,7 +1334,14 @@ func (s *svc) statAcrossProviders(ctx context.Context, req *provider.StatRequest
                }
                if resp.Info != nil {
                        info.Size += resp.Info.Size
-                       info.Mtime = utils.LaterTS(info.Mtime, resp.Info.Mtime)
+                       if utils.TSToUnixNano(resp.Info.Mtime) > utils.TSToUnixNano(info.Mtime) {
+                               info.Mtime = resp.Info.Mtime
+                               info.Etag = resp.Info.Etag
+                               info.Checksum = resp.Info.Checksum
+                       }
+                       if info.Etag == "" && info.Etag != resp.Info.Etag {
+                               info.Etag = resp.Info.Etag
+                       }
                }
        }
 
@@ -1705,7 +1712,14 @@ func (s *svc) listContainerAcrossProviders(ctx context.Context, req *provider.Li
                                }
                                // TODO(ishank011): aggregrate properties such as etag, checksum, etc.
                                p.Size += info.Size
-                               p.Mtime = utils.LaterTS(p.Mtime, info.Mtime)
+                               if utils.TSToUnixNano(info.Mtime) > utils.TSToUnixNano(p.Mtime) {
+                                       p.Mtime = info.Mtime
+                                       p.Etag = info.Etag
+                                       p.Checksum = info.Checksum
+                               }
+                               if p.Etag == "" && p.Etag != info.Etag {
+                                       p.Etag = info.Etag
+                               }
                                p.Type = provider.ResourceType_RESOURCE_TYPE_CONTAINER
                                p.MimeType = "httpd/unix-directory"
                        } else {
```

right @ishank011 ?

Anyway, we should add the expectations to this PRs feature. Hopefully, it can be mimicked using the localfs.

**Comment by butonic** [comment]
without the above diff I get
```
      WebDavPropertiesContext::theseEtagsShouldHaveChanged
      The etag '' of element '/' of user 'einstein' did not change.
      The etag '' of element '/virtual' of user 'einstein' did not change.
      The etag '"0330f2019a2524fc51bc62964a186ac4"' of element '/virtual/b' of user 'einstein' did not change.
      Failed asserting that 3 matches expected 0.
```

with it I can get rid of the last etag not changing:
```
      WebDavPropertiesContext::theseEtagsShouldHaveChanged
      The etag '' of element '/' of user 'einstein' did not change.
      The etag '' of element '/virtual' of user 'einstein' did not change.
      Failed asserting that 2 matches expected 0.
```

**Review by individual-it — CHANGES_REQUESTED** [review]
are these tests reva specific or should they be moved/copied to ocis?

File: tests/acceptance/features/apiVirtualViews/virtualViews.feature [hunk 1]
```diff
@@ -0,0 +1,65 @@
+@api @virtual-views-required
+Feature: virtual views
+  As admin
+  I want to be able to shard large folders over multiple storage providers
+  So that I can scale large numbers of users better.
+
+  Background:
+    Given user "einstein" deletes everything from folder "virtual/" using the WebDAV API
+    Given user "einstein" has created the following folders
```
[the remark this answers is not in the document]
Comment (butonic) [reply tests/acceptance/features/apiVirtualViews/virtualViews.feature:9]: I tonly happens when users are created. But for this suite I am using the demo users. So, I need to manually restore the initial virtual folders in the background step.

**Comment by butonic** [comment]
> are these tests reva specific or should they be moved/copied to ocis?

They are reva specific. The 'virtual views' feature allows *merging* multiple storage providers to serve a single folder. Not something owncloud 10 can do.

File: tests/acceptance/features/apiVirtualViews/virtualViews.feature (hunk 1 above)
[the remark this answers is not in the document]
Comment (individual-it) [reply tests/acceptance/features/apiVirtualViews/virtualViews.feature]: Any reason not to create "Alice" and delete the let the test suite delete the end?

**Comment by butonic** [comment]
@ishank011 I think we have a passing test! 🤞

**Review by ishank011 — APPROVED** [review]

**Review by C0rby — APPROVED** [review]

**Review by individual-it — COMMENTED** [review]
I would suggest not to use "einstein" as user in the tests but create a new user as we do everywhere else

File: tests/acceptance/features/apiVirtualViews/virtualViews.feature [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    And the last DAV response for user "einstein" should contain these nodes
+      | name |
+      | virtual/a    |
+      | virtual/b    |
+      | virtual/c    |
+      | virtual/k    |
+      | virtual/l    |
+      | virtual/z    |
+    And the last DAV response for user "einstein" should not contain these nodes
+      | name                              |
```
Comment (individual-it) [inline tests/acceptance/features/apiVirtualViews/virtualViews.feature:42]: table formatting could be improved

**Comment by butonic** [comment]
> I would suggest not to use "einstein" as user in the tests but create a new user as we do everywhere else
I did not want to make an ldap server necessary as I don't want to test managing users. I would actually like to reduce this test to a point where we only register the storage providers and don't even have users ... but I would have to test further if disabling user home creation on login is enough.

**Comment by butonic** [comment]
hm ... we could change the config of the default test suite to start two storage providers for /users .... but that would make the setup even more complex ...

**Review by ishank011 — APPROVED** [review]

**Pull request merged**